src/io/flutter/actions/ActionWithAnalytics.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.actions;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import io.flutter.FlutterInitializer;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * An action that sends analytics when performed.
+ */
+public abstract class ActionWithAnalytics extends AnAction {
+
+  /**
+   * Template method.  Implement @performAction.
+   */
+  @Override
+  public final void actionPerformed(AnActionEvent e) {
```

Comment:
Final methods often end up causing unexpected problems. It would be better to omit that modifier.

Reply:
👍 In general I hear you.  In this case though it's a template method and I don't want it accidentally overridden since the analytics call won't happen unless they think to call super.

Reply:
`@OverridingMethodsMustInvokeSuper` will do what you want and won't do what I don't want.

<img width="400" alt="screen shot 2018-05-10 at 12 43 02 pm" src="https://user-images.githubusercontent.com/8518285/39890466-b7de988e-544f-11e8-8848-828afe2d7904.png">

You'll need to add a library dependency on jsr305 to enable that annotation, but I think it's worth doing. (Remember to add it to all relevant module definitions. I think there are two.)

In its current form, this method appears to contradict the instructions from JetBrains:
[https://www.jetbrains.org/intellij/sdk/docs/tutorials/action_system/working_with_custom_actions.html](https://www.jetbrains.org/intellij/sdk/docs/tutorials/action_system/working_with_custom_actions.html)

Reply:
I don't have an opinion here, but we may want to move discussion out of a PR; perhaps when you're in the office Monday Steve?

Reply:
Sure.  We can discuss and rejigger.